packages/pyright-internal/src/tests/samples/typeNarrowingIsinstance6.py

```diff
@@ -51,7 +51,7 @@ class ChildB1(ParentB[_T2]):
 
 def func4(var: ParentB[int]):
     if isinstance(var, ChildB1):
-        reveal_type(var, expected_text="ChildB1[int]")
+        reveal_type(var, expected_text="<subclass of ParentB[int] and ChildB1[float]>")
```

Comment:
This doesn't seem like a good change. If we already have a narrower type var, it seems like it shouldn't widen it.
Do you know an example where narrowing to `ChildB1[int]` would be unsafe?

If it's just an issue of being difficult to implement, I think this change probably isn't bad enough to be a blocker.